src/sarif_converter.ts

```diff
@@ -0,0 +1,180 @@
+import { IAnalysisResult, ISuggestion, IFileSuggestion } from './interfaces/analysis-result.interface';
+import { Log, ReportingConfiguration, ReportingDescriptor  } from "sarif"
+
+interface ISarifSuggestion extends IFileSuggestion {
+  id: string;
+  ruleIndex: number;
+  rule: ReportingDescriptor 
+  level: ReportingConfiguration.level;
+  text: string;
+  file: string;
+}
+interface ISarifSuggestions {
+  [suggestionIndex: number]: ISarifSuggestion;
+}
+class Sarif {
+  private analysisResults: IAnalysisResult;
+  private suggestions: ISarifSuggestions= {}
+
+  constructor(analysisResults: IAnalysisResult) {
```

Comment:
Conceptually speaking, there is no meaning in the "Sarif" class being dependant/built around a single instance of the analysisResults.